CSS "offset-position": Update support in Firefox116 and Chrome

dipikabh opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->

#### Summary

1. Firefox 115 now supports the [`offset-position`](https://developer.mozilla.org/en-US/docs/Web/CSS/offset-position) property behind the flag `layout.css.motion-path-offset-position.enabled`. I am working on creating the content PR to reflect the syntax changes on the property page.
Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1559232

2. The syntax for `offset-position` property has changed.
A new keyword (`normal`) has been added: https://drafts.fxtf.org/motion/#offset-position-property.
    - Firefox supports the new syntax
    - Chrome seems to support the `offset-position` property now but the new keyword is not yet supported.

#### Test results and supporting details

1. Screenshots
<details>
<summary>In Firefox 115 beta, behind the flag</summary>

![fx115-firefox-beta-behind-flag](https://github.com/mdn/browser-compat-data/assets/23019147/a408d28c-a960-4995-b930-2b9f69c95bba)

</details>

<details>
<summary>In Chrome beta (Version 115.0.5790.56 (Official Build) beta (x86_64)), experimental is enabled</summary>

![fx115-chrome-beta-flagonbydefault](https://github.com/mdn/browser-compat-data/assets/23019147/330ca05a-2480-4b0c-bdcc-5a3a338d7617)

</details>

<details>
<summary>In Chrome Canary</summary>

![fc115-chrome-canary](https://github.com/mdn/browser-compat-data/assets/23019147/dac1cbe0-6fb3-4dfb-b96b-0774825fe746)

</details>

2. WPT: https://wpt.fyi/results/css/motion/parsing/offset-position-parsing-valid.html?label=master&product=chrome-116%5Bexperimental%5D&product=chrome-115%5Bbeta%5D&product=firefox-115%5Bexperimental%5D&product=safari%5Bbeta%5D

#### Questions

1. Do I need to indicate in some way that the Chrome version in BCD is the beta version?
2. Canary does support the `normal` keyword in syntax. But both Canary and Beta versions don't seem to support keyword `<position>` values like `bottom`, `left`, `right`. Should I be adding a note to that effect? Should I add Canary instead of Beta info to the BCD table?
3. Not sure if I should retain the chrome bug link: https://crbug.com/638055. I am assuming there would be another bug (that I could not find) to track the syntax update.

#### Related issues

Doc tracker for this update: https://github.com/mdn/content/issues/27174

queengooborg commented on css/properties/offset-position.json:
```diff
@@ -7,14 +7,27 @@
-              "version_added": false,
-              "notes": "See <a href='https://crbug.com/638055'>bug 638055</a>."
+              "version_added": "115",
+              "notes": "The keyword `normal` is not part of the syntax.",
```
Rather than add a note to Chrome, we should create a subfeature for this value, especially as Firefox supports it.

dipikabh commented:
Thanks a lot @queengooborg for the review and suggestion. I have now added a subfeature for the `normal` value (currently not supported in Chrome and Safari TP).
Also, this feature got moved from Fx115 to Fx116. I've updated the BCD table and this PR title to reflect this.

Elchi3 reviewed (commented):
Thanks for making this a sub feature!

BCD is all about "support" so we don't repeat that in notes or in feature ids.

dipikabh commented:
Thanks for reviewing and for the hints, @Elchi3!
